apis/pipelines/v1alpha5/provider_types.go

```diff
@@ -27,12 +27,8 @@ type Provider struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   ProviderSpec   `json:"spec,omitempty"`
-	Status ProviderStatus `json:"status,omitempty"`
-}
-
-type ProviderStatus struct {
-	Conditions Conditions `json:"conditions,omitempty"`
+	Spec   ProviderSpec `json:"spec,omitempty"`
+	Status Status       `json:"status,omitempty"`
```

Comment:
We previously used a dedicated `ProviderStatus` struct for the provider CRD status, but we didn't see the point in it because:

- It only has a single field, `Conditions`, which is part of the default `Status` struct already
- We don't set the provider status anywhere (except for tests, just so they compile)

So we removed `ProviderStatus` and just used the default `Status` struct.